### PR TITLE
Remove unused code + refactor after Kubeops removal

### LINKED ISSUE #5284: Reorganize and clean up code after Kubeops removal
**Summary**
Remove old dead code related mainly to `kubeops` and pre-plugins architecture.
Migrate code under `/pkg` that is directly related to `helm` plugin only.

**Description**
After removing `kubeops` (#5253) there is associated source code placed in the `/pkg` folder.
It needs reviewing and cleaning up the parts not needed. 

**Acceptance criteria**
Source code under `/pkg` that is no longer used has been removed.
Source code that is only related and used to e.g. `helm` plugin must be moved to that plugin.

### PR DESCRIPTION
**Pull request by castelblanque**
Signed-off-by: Rafa Castelblanque <rcastelblanq@vmware.com>

### Description of the change

After removing `kubeops` #5253, there was a lot of old useless backend code left in the Go modules.
This PR removes unused duplicate code related mainly to `kubeops` and pre-plugins age.
Also refactors some pieces, code under `/pkg` that is directly related to helm plugin only.

### Benefits

Simpler and cleaner code base.

### Possible drawbacks

N/A

### Applicable issues

- fixes #5284

### CODE REVIEW

**Review by beni0888 — APPROVED** [review]
LGTM! I've added several suggestions, most of them probably not directly related to the scope of this PR, as they likely refer to moved code, and anyway, none of them is blocking. So feel free to ignore them.

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	return f.response, f.err
+}
+
+func TestNonOCIValidate(t *testing.T) {
+	validRequest, err := http.NewRequest("GET", "http://example.com/index.yaml", strings.NewReader(""))
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	testCases := []struct {
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_validation_test.go:40]: That looks like a good improvement to me, we could start doing it!

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_validation_test.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+func (f *fakeHTTPCli) Do(r *http.Request) (*http.Response, error) {
+	f.request = r
+	return f.response, f.err
+}
+
+func TestNonOCIValidate(t *testing.T) {
+	validRequest, err := http.NewRequest("GET", "http://example.com/index.yaml", strings.NewReader(""))
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_validation_test.go:38]: Agree, it is a cleaner way.
Most part of the code is not new, so we have the traditional way of checking :)

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/utils/chart-client_test.go [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+}
+
+func TestGetChartHttp(t *testing.T) {
+	const repoName = "foo-repo"
+	testCases := []struct {
+		name          string
+		chartVersion  string
+		userAgent     string
+		tarballURL    string
+		errorExpected bool
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/utils/chart-client_test.go:77]: Good catch, thanks!

**Comment by castelblanque** [comment]
@beni0888 Thanks for the thorough review. Some comments I haven't changed because it is actually previously existing code and didn't want to  mess with it too much given how big the PR is already.
We can tackle it in future PRs.

**Pull request merged**